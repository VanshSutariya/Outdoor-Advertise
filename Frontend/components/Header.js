import React, { useState } from "react";
import Link from "next/link";

const NavBar = () => {
  const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false);

  const handleMobileMenuToggle = () => {
    console.log(isMobileMenuOpen);
    setIsMobileMenuOpen(!isMobileMenuOpen);
  };

  return (
    <>
      <nav className="flex font-inter justify-between items-center px-4 bg-black tracking-wide">
        <div className='flex mr-9 flex-col items-center justify-center w-full"'>
          <Link href="/">
            <span className="sm:w-[7rem] w-[5rem] text-white text-lg font-bold flex items-center">
              <img src="/logo.png" alt="Logo" className="mr-2 h-[60px]" />
            </span>
          </Link>
        </div>
        <div
          className="md:hidden flex cursor-pointer max-w-sm"
          onClick={handleMobileMenuToggle}
        >
          <div className="flex-1 ml-auto">
            <span className="text-white text-4xl">&#9776;</span>
          </div>
        </div>
        <div className="container hidden mx-auto md:flex justify-between items-center">
          <div className="flex items-center space-x-2">
            <div className="hidden md:flex font-semibold space-x-4">
              <Link href="/">
                <span className="text-white">Home</span>
              </Link>
              <Link href="/about">
                <span className="text-white">About</span>
              </Link>
              <Link href="/services">
                <span className="text-white">Services</span>
              </Link>
              <Link href="/contact">
                <span className="text-white">Contact</span>
              </Link>
            </div>
          </div>
          <div className="md:flex hidden items-center font-semibold space-x-4">
            <button className="text-white bg-gray-600 hover:bg-gray-700 active:bg-gray-900 px-4 py-1 rounded">
              <Link href="/login">Login</Link>
            </button>
            <button className="text-white bg-gray-600 hover:bg-gray-700 active:bg-gray-900 px-4 py-1 rounded">
              <Link href="/register">Register</Link>
            </button>
          </div>

          {/* Mobile Navigation Dropdown */}
        </div>

        {isMobileMenuOpen && (
          <div className="md:hidden z-10 absolute rounded-sm opacity-95 shadow-sm shadow-slate-800 top-16 px-4 right-4 left-auto bg-[#ffffff]">
            <div className="px-4 py-2 space-y-2 text-base font-semibold">
              <Link href="/">
                <span className="block my-2 text-black cursor-pointer">
                  Home
                </span>
              </Link>
              <Link href="/about">
                <span className="block my-2 text-black cursor-pointer">
                  About
                </span>
              </Link>
              <Link href="/services">
                <span className="block my-2 text-black cursor-pointer">
                  Services
                </span>
              </Link>
              <Link href="/register">
                <span className="block my-2 text-black cursor-pointer">
                  Register
                </span>
              </Link>
              <Link href="/login">
                <button className="block my-2 text-black cursor-pointer">
                  Login
                </button>
              </Link>
            </div>
          </div>
        )}
      </nav>
    </>
  );
};

export default NavBar;
