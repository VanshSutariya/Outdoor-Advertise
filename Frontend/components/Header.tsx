import React, { useEffect, useState } from "react";
import Link from "next/link";
import { useDispatch, useSelector } from "react-redux";
import { RootState } from "../store";
import { loginIn, logout } from "../store/auth-slice";
import { decode } from "jsonwebtoken";
import fetchUser from "../utils/http";
const NavBar: React.FC = () => {
  const dispatch = useDispatch();
  const [isMobileMenuOpen, setIsMobileMenuOpen] = useState<boolean>(false);

  const { user, isLoggedIn }: { user: string | null; isLoggedIn: boolean } =
    useSelector((state: RootState) => state.auth);

  useEffect(() => {
    const jwtCookie = document.cookie
      .split("; ")
      .find((row) => row.startsWith("jwt="));

    if (jwtCookie) {
      const token = jwtCookie.split("=")[1];
      const decodedToken = decode(token) as { id: string };
      if (decodedToken) {
        const fetchUserFunc = async () => {
          const userName = await fetchUser(decodedToken.id);
          dispatch(loginIn(userName));
        };
        fetchUserFunc();
      }
    }
  }, [dispatch]);

  const handleLogout = async () => {
    document.cookie = "jwt=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/;";
    dispatch(logout());
  };

  const handleMobileMenuToggle = () => {
    setIsMobileMenuOpen(!isMobileMenuOpen);
  };

  return (
    <>
      <nav className="flex justify-between items-center px-4 pb-2 tracking-wide border-b-[1px]">
        <div className="flex items-center">
          <Link href="/" className="flex items-center">
            <span className="w-full text-black text-lg font-bold  items-center">
              <img
                src="/logo.png"
                alt="Logo"
                className="w-full md:w-[120px] h-[70px]"
              />
            </span>
            <span className=" w-full px-3 mr-5 font-bold md:text-xl text-blue-950">
              OutdoorAdvertise
            </span>
          </Link>
        </div>
        <div
          className="md:hidden flex cursor-pointer max-w-sm"
          onClick={handleMobileMenuToggle}
        >
          {isLoggedIn && user && (
            <div className="  sm:flex mr-4 md:hidden items-center font-semibold space-x-4">
              <div className='flex flex-col items-center justify-center w-full"'>
                <Link href="/">
                  <img
                    src="/shopping-cart.png"
                    alt="Logo"
                    className=" h-[50px]"
                  />
                </Link>
              </div>
            </div>
          )}
          <div className="flex-1 ml-auto">
            <span className="text-black text-4xl">&#9776;</span>
          </div>
        </div>
        <div className="container hidden mx-auto md:flex justify-between items-center">
          <div className="flex items-center space-x-2">
            <div className="hidden md:flex pt-1 font-thin text-lg space-x-4">
              <Link href="/">
                <span className="text-black">Home</span>
              </Link>
              <Link href="/about">
                <span className="text-black">About</span>
              </Link>
              <Link href="/services">
                <span className="text-black">Services</span>
              </Link>
              <Link href="/contact">
                <span className="text-black">Contact</span>
              </Link>
            </div>
          </div>
          {!isLoggedIn && (
            <div className="md:flex hidden items-center font-semibold space-x-4">
              <button className="text-black  border-2 border-black hover:bg-gray-200 active:bg-gray-300 px-4 py-2 rounded-3xl">
                <Link href="/login">Login</Link>
              </button>
              <button className="text-black  border-2 border-black hover:bg-gray-200 active:bg-gray-300 px-4 py-2 rounded-3xl">
                <Link href="/register">Register</Link>
              </button>
            </div>
          )}
          {isLoggedIn && user && (
            <div className="md:flex hidden items-center font-semibold space-x-4">
              <div className='flex flex-col items-center justify-center w-full"'>
                <Link href="/">
                  <img
                    src="/shopping-cart.png"
                    alt="Logo"
                    className=" h-[50px]"
                  />
                </Link>
              </div>
              <p className="text-xl text-black">{user}</p>
              <button
                onClick={handleLogout}
                className="text-white bg-black hover:bg-gray-700 active:bg-gray-900 px-4 py-2 rounded-3xl"
              >
                <Link href="/">LogOut</Link>
              </button>
            </div>
          )}

          {/* Mobile Navigation Dropdown */}
        </div>

        {isMobileMenuOpen && (
          <>
            <div className="md:hidden z-10 absolute rounded-sm opacity-95 shadow-sm shadow-slate-800 top-16 px-4 right-4 left-auto bg-[#ffffff]">
              <div className="px-4 py-2 space-y-2 text-base font-semibold">
                <Link href="/">
                  <span className="block my-2 text-black cursor-pointer">
                    Home
                  </span>
                </Link>
                <Link href="/about">
                  <span className="block my-2 text-black cursor-pointer">
                    About
                  </span>
                </Link>
                <Link href="/services">
                  <span className="block my-2 text-black cursor-pointer">
                    Services
                  </span>
                </Link>
                {!isLoggedIn && (
                  <>
                    <Link href="/register">
                      <span className="block my-2 text-black cursor-pointer">
                        Register
                      </span>
                    </Link>
                    <Link href="/login">
                      <button className="block my-2 text-black cursor-pointer">
                        Login
                      </button>
                    </Link>
                  </>
                )}
                {isLoggedIn && (
                  <>
                    <Link href="/">
                      <button className="block my-2 text-black cursor-pointer">
                        LogOut
                      </button>
                    </Link>
                    <Link href="/">
                      <button className="block my-2 text-black cursor-pointer">
                        Your Cart
                      </button>
                    </Link>
                  </>
                )}
              </div>
            </div>
          </>
        )}
      </nav>
    </>
  );
};

export default NavBar;
