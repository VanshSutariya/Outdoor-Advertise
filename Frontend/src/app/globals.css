@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@400;500;600;700;800;900&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Ubuntu:wght@500&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Inter:wght@100..900&display=swap");
@tailwind base;
@tailwind components;
@tailwind utilities;

::-webkit-scrollbar {
  width: 0px;
}

::-webkit-scrollbar-track {
  background: #f1f1f1;
}

::-webkit-scrollbar-thumb {
  background: #888;
  border-radius: 5px;
}

::-webkit-scrollbar-thumb:hover {
  background: #555;
}
