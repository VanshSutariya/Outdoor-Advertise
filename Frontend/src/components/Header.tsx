"use client";
import React, { useState } from "react";
import Link from "next/link";
import { FaUserLarge } from "react-icons/fa6";
import { RootState } from "@/store";
import { logout } from "@/store/auth-slice";
import { useRouter } from "next/navigation";
import { useDispatch, useSelector } from "react-redux";

const Header: React.FC = () => {
  const router = useRouter();
  const dispatch = useDispatch();
  const [isMobileMenuOpen, setIsMobileMenuOpen] = useState<boolean>(false);

  const {
    userId,
    userName,
    isLoggedIn,
    userRole,
  }: {
    userId: string | null;
    userName: string | null;
    isLoggedIn: boolean;
    userRole: string | null;
  } = useSelector((state: RootState) => state.auth);

  const handleLogout = async () => {
    document.cookie = "jwt=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/";
    dispatch(logout());
  };

  const handleMobileMenuToggle = () => {
    setIsMobileMenuOpen(!isMobileMenuOpen);
  };

  function handleAccountPage() {
    router.push("/outdoorAd/account");
  }

  return (
    <>
      <nav className="flex justify-between items-center px-4 pb-2 tracking-wide border-b-[1px]">
        <div data-testid="headertest" className="flex items-center">
          <Link href="/outdoorAd" className="flex items-center">
            <img
              src="/logo.png"
              alt="logo"
              className="w-full md:w-[130px] h-[60px]"
            />

            <span className=" w-full px-3 mr-5 font-bold md:text-xl text-blue-950">
              OutdoorAdvertise
            </span>
          </Link>
        </div>
        <div
          className="md:hidden flex cursor-pointer max-w-sm"
          onClick={handleMobileMenuToggle}
        >
          {isLoggedIn && userId && (
            <div className="sm:flex mr-4 md:hidden items-center font-semibold space-x-4">
              <div className='flex flex-col items-center justify-center w-full"'>
                <Link href="/outdoorAd/cart">
                  <img
                    src="/shopping-cart.png"
                    alt="shoppingCart"
                    className=" h-[35px]"
                  />
                </Link>
              </div>
            </div>
          )}
          <div className="flex-1 ml-auto">
            <span className="text-black text-4xl">&#9776;</span>
          </div>
        </div>
        <div className="container hidden mx-auto md:flex justify-between items-center">
          <div className="flex items-center space-x-2">
            <div className="hidden md:flex pt-1 font-inter  text-lg space-x-4">
              <Link href="/outdoorAd">
                <span className="text-black">Home</span>
              </Link>
              <Link href="/outdoorAd/contactUs">
                <span className="text-black">Contact</span>
              </Link>
            </div>
          </div>
          {!isLoggedIn && (
            <div className="md:flex hidden items-center font-semibold space-x-4">
              <Link href="/outdoorAd/login">
                <button className="text-black  border-2 border-black hover:bg-gray-200 active:bg-gray-300 px-4 py-2 rounded-3xl">
                  Login
                </button>
              </Link>
              <Link href="/outdoorAd/register">
                <button className="text-black  border-2 border-black hover:bg-gray-200 active:bg-gray-300 px-4 py-2 rounded-3xl">
                  Register
                </button>
              </Link>
            </div>
          )}
          {isLoggedIn && userId && (
            <div className="md:flex hidden items-center font-semibold space-x-2">
              <Link href="/outdoorAd/cart">
                <button className="btn bg-transparent">
                  <img
                    src="/shopping-cart.png"
                    alt="Logo"
                    className=" h-[35px]"
                  />
                </button>
              </Link>
              {/* profile dropdown */}
              <div className=" dropdown dropdown-end">
                <div tabIndex={0} role="button" className="flex ">
                  <div className="flex items-center mt-2 border-2 p-2 hover:bg-gray-200 rounded-xl">
                    <div className=" items-center">
                      <p className="text-lg p-1">
                        {userName &&
                          userName.substring(0, 1).toUpperCase() +
                            userName.substring(1)}
                      </p>
                    </div>
                    <div className="p-1">
                      <FaUserLarge size={25} />
                    </div>
                  </div>
                </div>
                <ul
                  tabIndex={0}
                  className="mt-3 z-[1] p-2 shadow menu menu-sm dropdown-content bg-base-100 rounded-box "
                >
                  <li className=" items-center ">
                    <button onClick={handleAccountPage}>Account</button>
                  </li>
                  <li className=" items-center">
                    <Link href="/outdoorAd">
                      <button onClick={handleLogout}>Logout</button>
                    </Link>
                  </li>
                  {userRole && userRole !== "user" && (
                    <li className=" items-center">
                      <Link href="/outdoorAd/createPoster">Create Poster</Link>
                    </li>
                  )}
                  <li className=" items-center">
                    <Link href="/outdoorAd/updatePassword">UpdatePassword</Link>
                  </li>
                </ul>
              </div>
            </div>
          )}

          {/* Mobile Navigation Dropdown */}
        </div>

        {isMobileMenuOpen && (
          <>
            <div className="md:hidden z-10 absolute rounded-sm opacity-95 shadow-sm shadow-slate-800 top-16 px-4 right-4 left-auto bg-[#ffffff]">
              <div className="px-4 py-2 space-y-2 text-base font-semibold">
                <Link href="/outdoorAd">
                  <span className="block my-2 text-black cursor-pointer">
                    Home
                  </span>
                </Link>
                {!isLoggedIn && (
                  <>
                    <Link href="/outdoorAd/register">
                      <span className="block my-2 text-black cursor-pointer">
                        Register
                      </span>
                    </Link>
                    <Link href="/outdoorAd/login">
                      <button className="block my-2 text-black cursor-pointer">
                        Login
                      </button>
                    </Link>
                  </>
                )}
                {isLoggedIn && (
                  <>
                    <Link href="/outdoorAd">
                      <button
                        onClick={handleLogout}
                        className="block my-2 text-black cursor-pointer"
                      >
                        LogOut
                      </button>
                    </Link>
                    <Link href="/outdoorAd/account">
                      <button className="block my-2 text-black cursor-pointer">
                        Account
                      </button>
                    </Link>
                    <Link href="/outdoorAd/updatePassword">
                      <button className="block my-2 text-black cursor-pointer">
                        Update Password
                      </button>
                    </Link>
                    {userRole && userRole !== "user" && (
                      <Link href="/outdoorAd/createPoster">
                        <button className="block my-2 text-black cursor-pointer">
                          Create Poster
                        </button>
                      </Link>
                    )}
                  </>
                )}
              </div>
            </div>
          </>
        )}
      </nav>
    </>
  );
};

export default Header;
