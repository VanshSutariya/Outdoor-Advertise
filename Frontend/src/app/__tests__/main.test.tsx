import React, { ReactElement } from "react";
import {
  render,
  screen,
  fireEvent,
  RenderResult,
  waitFor,
} from "@testing-library/react";
import SignIn from "@/components/auth-login-regis/login";
import Providers from "@/store/provider";
import { describe, it } from "@jest/globals";
import Header from "@/components/Header";
import PosterForm from "@/components/CreatePoster/posterForm";
import fetchMock from "jest-fetch-mock";

const sendComponent = (component: ReactElement): RenderResult =>
  render(<Providers> {component}</Providers>);

jest.mock("next/router", () => require("next-router-mock"));
jest.mock("node-fetch", () => require("jest-fetch-mock"));
jest.mock("next/navigation", () => ({
  ...require("next-router-mock"),
  useSearchParams: () => jest.fn(),
}));

describe("Main Home Page", () => {
  it("renders Header Page", () => {
    const { container } = sendComponent(<Header />);
    expect(container).toMatchSnapshot();
  });
});

// login page test cases
describe("SignIn Component", () => {
  it("renders the login form", () => {
    sendComponent(<SignIn />);
    expect(screen.getByTestId("emailtest")).toBeInTheDocument();
  });

  it("redirects on successful login", async () => {
    sendComponent(<SignIn />);
    expect(screen.getByTestId("emailtest")).toBeInTheDocument();
    expect(screen.getByTestId("passwordtest")).toBeInTheDocument();
    expect(screen.getByTestId("loginSubmit")).toBeEnabled();

    await waitFor(() => {
      fireEvent.change(screen.getByTestId("emailtest"), {
        target: { value: "sutariyavansh@gmail.com" },
      });
      fireEvent.change(screen.getByTestId("passwordtest"), {
        target: { value: "74108520" },
      });

      fireEvent.submit(screen.getByTestId("loginSubmit"));
    });
  });

  it("displays validation errors for empty fields", async () => {
    sendComponent(<SignIn />);
    fireEvent.submit(screen.getByTestId("loginSubmit"));

    await waitFor(() => {
      expect(screen.getByText("Email is required")).toBeInTheDocument();
      expect(screen.getByText("Password is required")).toBeInTheDocument();
    });
  });

  it("displays validation error for invalid email format", async () => {
    sendComponent(<SignIn />);
    fireEvent.change(screen.getByTestId("emailtest"), {
      target: { value: "invalidemail" },
    });
    fireEvent.change(screen.getByTestId("passwordtest"), {
      target: { value: "password" },
    });
    fireEvent.submit(screen.getByTestId("loginSubmit"));

    await waitFor(() => {
      expect(screen.getByText("Email is invalid")).toBeInTheDocument();
    });
  });
});

describe("CreatePoster Page", () => {
  it("renders the CreatePoster form", () => {
    const { container } = sendComponent(<PosterForm />);
    expect(container).toMatchSnapshot();
  });

  it("displays validation errors for invalid form inputs", async () => {
    sendComponent(<PosterForm />);

    fireEvent.submit(screen.getByTestId("createPosterForm"));

    await waitFor(() => {
      expect(screen.getByText("Title must be required")).toBeInTheDocument();
    });

    fireEvent.change(screen.getByLabelText("Title"), {
      target: { value: "" },
    });

    fireEvent.submit(screen.getByTestId("createPosterForm"));

    await waitFor(() => {
      expect(screen.getByText("Title must be required")).toBeInTheDocument();
    });
  });
});
