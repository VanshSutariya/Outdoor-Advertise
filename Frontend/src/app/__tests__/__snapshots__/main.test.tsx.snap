// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePoster Page renders the CreatePoster form 1`] = `
<div>
   
  <div>
    <a
      class=" text-lg md:px-[60px] pt-3 flex text-blue-600 hover:text-blue-700 active:text-blue-500 "
      href="/outdoorAd/account"
    >
      <p
        class="hover:scale-110 active:scale-95 duration-150 flex"
      >
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M216.4 163.7c5.1 5 5.1 13.3.1 18.4L155.8 243h231.3c7.1 0 12.9 5.8 12.9 13s-5.8 13-12.9 13H155.8l60.8 60.9c5 5.1 4.9 13.3-.1 18.4-5.1 5-13.2 5-18.3-.1l-82.4-83c-1.1-1.2-2-2.5-2.7-4.1-.7-1.6-1-3.3-1-5 0-3.4 1.3-6.6 3.7-9.1l82.4-83c4.9-5.2 13.1-5.3 18.2-.3z"
          />
        </svg>
         Back
      </p>
    </a>
  </div>
  <main
    class=" pb-5"
  >
    <div
      class=" flex justify-center items-center mb-6"
    >
      <form
        class="w-[800px] border-[2px] border-gray-100 shadow-xl rounded-xl md:p-4 "
        data-testid="createPosterForm"
      >
        <div
          class="flex justify-center items-center pb-[5px] md:-mt-3 "
        >
          <h1
            class="text-4xl mt-3 tracking-wide md:pl-5 font-inter text-center md:w-full "
          >
            Create Your Poster
          </h1>
          <div
            class=" flex  justify-end md:pr-12"
          >
            <img
              alt=""
              class="h-20 w-20   "
              src="/billboard.png"
            />
          </div>
        </div>
        <div
          class=" flex flex-wrap -mx-3 mb-6"
        >
          <div
            class="w-full md:w-2/5 px-3"
          >
            <label
              class="block  tracking-wide font-poppins text-sm font-bold mb-2"
              for="title"
            >
              Title
            </label>
            <input
              class="appearance-none font-poppins block w-full bg-slate-100 hover:bg-slate-200 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-200 focus:border-gray-100"
              id="title"
              min="1"
              name="title"
              placeholder="shivranjini, ahmedabad"
              type="text"
              value=""
            />
          </div>
          <div
            class="w-full md:w-[240px] px-3 mb-6 md:mb-0"
          >
            <label
              class="block tracking-wide font-poppins text-sm font-bold mb-2"
              for="mediatype"
            >
              MediaType
            </label>
            <div
              class="relative"
            >
              <select
                class="appearance-none font-poppins block w-full bg-slate-100 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-200 focus:border-gray-100"
                id="mediatype"
                name="mediatype"
              >
                <option
                  value="Poles"
                >
                  Poles
                </option>
                <option
                  value="Airports"
                >
                  Airports
                </option>
                <option
                  value="RailwayPlatforms"
                >
                  RailwayPlatforms
                </option>
                <option
                  value="Rickshaws"
                >
                  Rickshaws
                </option>
                <option
                  value="Billboard"
                >
                  Billboard
                </option>
                <option
                  value="BusStands"
                >
                  BusStands
                </option>
                <option
                  value="ShoppingMalls"
                >
                  ShoppingMalls
                </option>
                <option
                  value="Footoverbridges"
                >
                  Footoverbridges
                </option>
                <option
                  value="Buildings"
                >
                  Buildings
                </option>
                <option
                  value="Buses"
                >
                  Buses
                </option>
              </select>
              <div
                class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-black"
              >
                <svg
                  class="fill-current h-4 w-4"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="w-full md:w-[230px] px-3 mb-6 md:mb-0"
          >
            <label
              class="block tracking-wide font-poppins text-sm font-bold mb-2"
              for="lightingType"
            >
              LightingType
            </label>
            <div
              class="relative"
            >
              <select
                class="appearance-none font-poppins block w-full bg-slate-100 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-200 focus:border-gray-100"
                id="lightingType"
                name="lightingType"
              >
                <option
                  value="FRONT-LIGHTING"
                >
                  FRONT-LIGHTING
                </option>
                <option
                  value="BACK-LIGHTING"
                >
                  BACK-LIGHTING
                </option>
                <option
                  value="LED-SCREEN"
                >
                  LED-SCREEN
                </option>
                <option
                  value="NO-LIGHTING"
                >
                  NO-LIGHTING
                </option>
              </select>
              <div
                class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-black"
              >
                <svg
                  class="fill-current h-4 w-4"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class=" flex flex-wrap -mx-3 mb-6"
        >
          <div
            class="w-full md:w-1/2 px-3 mb-6 md:mb-0"
          >
            <label
              class="block  tracking-wide font-poppins text-sm font-bold mb-2"
              for="minQty"
            >
              Minimum Poles
            </label>
            <input
              class="appearance-none font-poppins block w-full bg-slate-100 hover:bg-slate-200 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-200 focus:border-gray-100"
              id="minQty"
              min="1"
              name="minQty"
              placeholder="1"
              type="number"
              value="1"
            />
          </div>
          <div
            class="w-full md:w-1/2 px-3"
          >
            <label
              class="block  tracking-wide font-poppins text-sm font-bold mb-2"
              for="maxQty"
            >
              Maximum Poles
            </label>
            <input
              class="appearance-none font-poppins block w-full bg-slate-100 hover:bg-slate-200 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-200 focus:border-gray-100"
              id="maxQty"
              min="1"
              name="maxQty"
              placeholder="10"
              type="number"
              value="1"
            />
          </div>
        </div>
        <div
          class=" flex flex-wrap -mx-3 mb-6"
        >
          <div
            class="w-full md:w-2/3 px-3 mb-6 md:mb-0"
          >
            <label
              class="block  tracking-wide font-poppins text-sm font-bold mb-2"
              for="landmark"
            >
              Landmark
            </label>
            <input
              class="appearance-none font-poppins block w-full bg-slate-100 hover:bg-slate-200 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-200 focus:border-gray-100"
              id="landmark"
              min="1"
              name="landmark"
              placeholder="vesu cross road"
              type="text"
              value=""
            />
          </div>
          <div
            class="w-full md:w-1/3 px-3"
          >
            <label
              class="block  tracking-wide font-poppins text-sm font-bold mb-2"
              for="facingFrom"
            >
              Facing Form
            </label>
            <input
              class="appearance-none font-poppins block w-full bg-slate-100 hover:bg-slate-200 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-200 focus:border-gray-100"
              id="facingFrom"
              min="1"
              name="facingFrom"
              placeholder="phonewala"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="flex flex-wrap -mx-3 mb-3"
        >
          <div
            class="w-full px-3"
          >
            <div>
              <label
                class="block tracking-wide font-poppins text-sm font-bold mb-2"
                for="Address"
              >
                Address
              </label>
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                autocomplete="off"
                class="appearance-none font-poppins block w-full hover:bg-slate-200 bg-slate-100 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-100 focus:border-gray-100"
                data-testid="addressTest"
                disabled=""
                placeholder="vesu cross road"
                role="combobox"
                type="text"
                value=""
              />
              <div
                class="relative"
              >
                <div
                  class="absolute top-full left-0 right-0 bg-white border border-gray-300 z-10 max-h-52 overflow-y-auto"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-wrap -mx-3 mb-6"
        >
          <div
            class="w-full md:w-1/3 px-3 mb-6 md:mb-0"
          >
            <label
              class="block  tracking-wide font-poppins text-sm font-bold mb-2"
              for="state"
            >
              State
            </label>
            <input
              class="appearance-none font-poppins block w-full bg-slate-100 hover:bg-slate-200 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-200 focus:border-gray-100"
              id="state"
              min="1"
              name="state"
              placeholder="gujarat"
              type="text"
              value=""
            />
          </div>
          <div
            class="w-full md:w-1/3 px-3 mb-6 md:mb-0"
          >
            <label
              class="block  tracking-wide font-poppins text-sm font-bold mb-2"
              for="city"
            >
              City
            </label>
            <input
              class="appearance-none font-poppins block w-full bg-slate-100 hover:bg-slate-200 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-200 focus:border-gray-100"
              id="city"
              min="1"
              name="city"
              placeholder="ahmedabad"
              type="text"
              value=""
            />
          </div>
          <div
            class="w-full md:w-1/3 px-3 mb-6 md:mb-0"
          >
            <label
              class="block  tracking-wide font-poppins text-sm font-bold mb-2"
              for="mindays"
            >
              Min Days
            </label>
            <input
              class="appearance-none font-poppins block w-full bg-slate-100 hover:bg-slate-200 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-200 focus:border-gray-100"
              id="mindays"
              min="1"
              name="mindays"
              placeholder="5"
              type="number"
              value="2"
            />
          </div>
        </div>
        <div
          class="flex flex-wrap -mx-3 mb-6"
        >
          <div
            class="w-full md:w-1/3 px-3 mb-6 md:mb-0"
          >
            <label
              class="block  tracking-wide font-poppins text-sm font-bold mb-2"
              for="perDayPrice"
            >
              Per Day Price
            </label>
            <input
              class="appearance-none font-poppins block w-full bg-slate-100 hover:bg-slate-200 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-200 focus:border-gray-100"
              id="perDayPrice"
              min="1"
              name="perDayPrice"
              placeholder="5000"
              type="number"
              value="1"
            />
          </div>
          <div
            class="w-full md:w-1/3 px-3 mb-6 md:mb-0"
          >
            <label
              class="block  tracking-wide font-poppins text-sm font-bold mb-2"
              for="height"
            >
              Height In Foot
            </label>
            <input
              class="appearance-none font-poppins block w-full bg-slate-100 hover:bg-slate-200 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-200 focus:border-gray-100"
              id="height"
              min="1"
              name="height"
              placeholder="30"
              type="number"
              value="1"
            />
          </div>
          <div
            class="w-full md:w-1/3 px-3 mb-6 md:mb-0"
          >
            <label
              class="block  tracking-wide font-poppins text-sm font-bold mb-2"
              for="width"
            >
              Width In Foot
            </label>
            <input
              class="appearance-none font-poppins block w-full bg-slate-100 hover:bg-slate-200 border border-gray-100 rounded-md py-3 px-4 leading-tight focus:outline-none focus:bg-slate-200 focus:border-gray-100"
              id="width"
              min="1"
              name="width"
              placeholder="30"
              type="number"
              value="1"
            />
          </div>
        </div>
        <div
          class="flex  items-center font-poppins "
        >
          <div
            class="w-full md:w-3/4 justify-center mb-3 md:mb-0"
          >
            <div
              class="flex justify-center"
            >
              <input
                class="w-full hidden "
                id="file-upload"
                type="file"
              />
              <label
                class="bg-orange-400  cursor-pointer text-center hover:bg-orange-600  transition-all duration-500 font-bold text-lg text-white p-2  rounded-lg "
                for="file-upload"
              >
                Choose file
              </label>
            </div>
          </div>
          <div
            class="w-full md:w-1/4 flex justify-end pr-10 mb-3 md:mb-0"
          >
            <button
              class="px-5 py-2 rounded-xl text-xl text-white bg-blue-600 hover:bg-blue-700 active:bg-blue-800"
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
  </main>
</div>
`;

exports[`Main Home Page renders Header Page 1`] = `
<div>
   
  <nav
    class="flex justify-between items-center px-4 pb-2 tracking-wide border-b-[1px]"
  >
    <div
      class="flex items-center"
      data-testid="headertest"
    >
      <a
        class="flex items-center"
        href="/outdoorAd"
      >
        <img
          alt="logo"
          class="w-full md:w-[130px] h-[60px]"
          src="/logo.png"
        />
        <span
          class=" w-full px-3 mr-5 font-bold md:text-xl text-blue-950"
        >
          OutdoorAdvertise
        </span>
      </a>
    </div>
    <div
      class="md:hidden flex cursor-pointer max-w-sm"
    >
      <div
        class="flex-1 ml-auto"
      >
        <span
          class="text-black text-4xl"
        >
          ☰
        </span>
      </div>
    </div>
    <div
      class="container hidden mx-auto md:flex justify-between items-center"
    >
      <div
        class="flex items-center space-x-2"
      >
        <div
          class="hidden md:flex pt-1 font-inter  text-lg space-x-4"
        >
          <a
            href="/outdoorAd"
          >
            <span
              class="text-black"
            >
              Home
            </span>
          </a>
          <a
            href="/outdoorAd/contactUs"
          >
            <span
              class="text-black"
            >
              Contact
            </span>
          </a>
        </div>
      </div>
      <div
        class="md:flex hidden items-center font-semibold space-x-4"
      >
        <a
          href="/outdoorAd/login"
        >
          <button
            class="text-black  border-2 border-black hover:bg-gray-200 active:bg-gray-300 px-4 py-2 rounded-3xl"
          >
            Login
          </button>
        </a>
        <a
          href="/outdoorAd/register"
        >
          <button
            class="text-black  border-2 border-black hover:bg-gray-200 active:bg-gray-300 px-4 py-2 rounded-3xl"
          >
            Register
          </button>
        </a>
      </div>
    </div>
  </nav>
</div>
`;
